beacon-chain/rpc/eth/beacon/handlers.go

```diff
@@ -735,9 +755,21 @@ func (s *Server) publishBlockSSZ(ctx context.Context, w http.ResponseWriter, r *
 		httputil.HandleError(w, api.VersionHeader+" header is required", http.StatusBadRequest)
 		return
 	}
+	if !versionRequired && versionHeader == "" {
+		versionHeader = s.versionHeaderFromCurrentSlot()
```

Comment:
While the current slot should definitionally be equal to the slot of the published block, wouldn't it be ideal to base this on the slot of the block itself?

If the posted block is json, we can unmarshal into a narrow type just to read the slot. Then pick the version based on the block's slot.

```
type SlotPeeker struct {
  Block struct {
    Slot primitives.Slot `json:"slot"`
  } `json:"message"`
}

p := &SlotPeeker{}
if err := json.Unmarshal(body, p) {
  ...
}
```

If the posted block is ssz, it can be unmarshaled to the correct type using https://github.com/prysmaticlabs/prysm/blob/develop/encoding/ssz/detect/configfork.go#L53

Reply:
Yeah sure, add another wrapper for "block contents" which specifies the block field as `json.RawMessage`. If the deneb wrapper type is published, you'll have len() > 0 for that field; then unmarshal the `json.RawMessage` value into the `SlotPeeker`. If the field has len() == 0, try to unmarshal the original body into the `SlotPeeker`.